staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1008,12 +1008,14 @@ func (e *Store) DeleteCollection(ctx context.Context, deleteValidation rest.Vali
 	}
 	// Spawn a number of goroutines, so that we can issue requests to storage
 	// in parallel to speed up deletion.
-	// TODO: Make this proportional to the number of items to delete, up to
+	// It is proportional to the number of items to delete, up to
 	// DeleteCollectionWorkers (it doesn't make much sense to spawn 16
 	// workers to delete 10 items).
 	workersNumber := e.DeleteCollectionWorkers
 	if workersNumber < 1 {
 		workersNumber = 1
+	} else if workersNumber > len(items) {
+		workersNumber = len(items)
```

Comment:
what happens with zero items/workers?

Reply:
it's probably fine, since there isn't anything to delete then

But I would probably feel safer maxing it with 1,
So basically:
max(1, min(len(items), e.DeleteCollectionWorkers)

Reply:
Agreed, it will at least clarify the case... I'll make the change

Reply:
ptal

I moved the if around to take care of the 0 items case.

---

Also, do you think it is a good idea to just return up front if there are no items in the collection (I don't know if this can even happen)?

I added a check to just return successfully if there are no items in the collection, instead of going through the WaitGroup and workers code which ultimately won't do anything.